socorro/cron/jobs/ftpscraper.py

```diff
@@ -71,25 +80,46 @@ def patient_urlopen(url, max_attempts=4, sleep_time=20):
 
 class ScrapersMixin(object):
 
-    def getLinks(self, url, startswith=None, endswith=None):
+    def get_links(self, url, startswith=None, endswith=None):
 
         html = ''
         results = []
         content = patient_urlopen(url, sleep_time=30)
         if not content:
             return []
+
         html = lxml.html.document_fromstring(content)
 
-        for element, attribute, link, pos in html.iterlinks():
-            if startswith:
+        def url_match(link):
+            # The link might be something like "/pub/mobile/nightly/"
+            # but we're looking for a path that starts with "nightly".
+            # So first we need to remove what's part of the base URL
+            # to make a fair comparison.
+            if startswith is not None:
+                # If the current URL is http://example.com/some/dir/
+                # and the link is /some/dir/mypage/ and the thing
+                # we're looking for is "myp" then this should be true
+                path = urlparse.urlparse(url).path
+                if link.startswith(path):
+                    link = link.replace(path, '')
```

Comment:
this substitution doesn't persist outside the method and I think that results in an incorrect final result.

Let's say we use your example:
```
url is "http://example.com/some/dir/"
path is then "/some/dir/"
the link is "/some/dir/mypage/"
start_with is "myp"
so we get a match and True is returned.
then down in your loop, the link "/some/dir/mypage" is accepted
and gets married with the url to form:
"http://example.com/some/dir/some/dir/mypage"
```
I don't think that's correct